pandas/indexes/multi.py

```diff
@@ -1007,6 +1008,9 @@ def from_tuples(cls, tuples, sortorder=None, names=None):
             # I think this is right? Not quite sure...
             raise TypeError('Cannot infer number of levels from empty list')
 
+        if not _check_equal_length(tuples):
```

Comment:
`to_object_array_tuples` seems to deliberately find the length of the longest inner sequence to pad everything to.  It is used indirectly by DataFrame.  If `to_object_array_tuples` were to throw an error on unbalanced input, would that affect DataFrame's behavior of padding things with NaN in some cases?